spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/GravitinoIcebergCatalog.java

```diff
@@ -85,4 +92,30 @@ public Identifier[] listFunctions(String[] namespace) throws NoSuchNamespaceExce
   public UnboundFunction loadFunction(Identifier ident) throws NoSuchFunctionException {
     return ((SparkCatalog) sparkCatalog).loadFunction(ident);
   }
+
+  @Override
+  public Procedure loadProcedure(Identifier identifier) throws NoSuchProcedureException {
+    String[] namespace = identifier.namespace();
```

Comment:
why not directly using `((SparkCatalog) sparkCatalog).loadProcedure()`?

Reply:
> why not directly using `((SparkCatalog) sparkCatalog).loadProcedure()`?

https://github.com/apache/iceberg/blob/ea916c1700ef37377d3e8bba1197b3fe32d1a248/spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java#L121

to avoid c`atalogAndIdentifier.catalog()` do not equal to `tableCatalog`

Reply:
why they are not equals, and how you avoid it?

Reply:
and please add some description

Reply:
> why they are not equals, and how you avoid it?
1. The `GravitinoIcebergCatalog` does not extend the `SparkCatalog`, and it cannot extend the `SparkCatalog` because the  `initialize` method of `SparkCatalog` is final and cannot be overrided.
2. if directly using `((SparkCatalog) sparkCatalog).loadProcedure()`, the `tableCatalog` is `SparkCatalog` but `catalogAndIdentifier.catalog()` is `GravitinoIcebergCatalog`, and it will cause the [check](https://github.com/apache/iceberg/blob/ea916c1700ef37377d3e8bba1197b3fe32d1a248/spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java#L121) failed.

![image](https://github.com/datastrato/gravitino/assets/94670132/2d076d95-dcdf-4685-8b8a-483ebcc4d034)

Reply:
ok

Reply:
cloud you add comment like this?
```
Proceduers will validate the equality of the catalog registered to Spark catalogManager and the catalog passed to `ProcedureBuilder`which invokes loadProceduer().  To meet the requirement , override the method to pass `GravitinoIcebergCatalog` to the `ProcedureBuilder` instead of the internal spark catalog.
```

Reply:
updated.